### ISSUE TITLE
Deprecate `CheckBox` so that all `Element` objects are a subclass of `Text`

### ISSUE THREAD

**Issue by MthwRobinson** [Unstructured-IO/unstructured#3053]
The goal of this issue is to deprecate the `CheckBox` element type, leaving all remaining `Element` classes as subclasses of `Text`. Early in the library the potential for non-text `Element` types was envisioned, but as the library evolved everything wound u being a subclass of `Text`.

**Comment by hubert-rutkowski85** [comment]
Also a bit related (but probably deserves a new issue?), we should move the `category` field to Element, so that linters will not complain. 
![image](https://github.com/Unstructured-IO/unstructured/assets/157481729/436db668-39f7-4c25-9380-9cff6f9ddf13)

**Comment by MthwRobinson** [comment]
Yeah agree on the `category` element, @hubert-rutkowski85 could you open a separate issue for that?

**Comment by hubert-rutkowski85** [comment]
Sure! Created here https://github.com/Unstructured-IO/unstructured/issues/3055 and started prototyping it.

**Comment by 9876543210-tc-0123456789** [comment]
Hi @MthwRobinson @scanny — I'd like to pick this up. 
I've checked the current state and it looks like some groundwork is already done (#2253 moved `text` to `Element`, #3056 moved `category` to `Element`), but `CheckBox` still inherits directly from `Element` with no deprecation.


**Before I start, I'd like to confirm the preferred approach:**

**Option A — Rebase `CheckBox` onto `Text`:** Change `class CheckBox(Element)` to `class CheckBox(Text)`, keep the `checked` attribute, default `text=""`. Minimal diff, backward-compatible for anyone doing `isinstance(el, CheckBox)`, but leaves a Text subclass with no meaningful text content.

**Option B — Remove `CheckBox` entirely:** Replace with a generic `Text` element (e.g. `text="Checked"` / `text="Unchecked"`) and store the boolean `checked` state in `ElementMetadata`. Cleaner long-term, but a breaking change for anyone matching on `type(el).__name__ == "CheckBox"` or deserializing old JSON.

Which direction do you prefer?
